Added new page for automated analysis, as well as screenshots (#29406)
DataDog/documentation: content/en/profiler/automated_analysis.md

---

> title: Automated Analysis

Reviewer: High-level question, but what's the reason we call this Automated Analysis instead of Recommendations like we have for [APM](https://github.com/DataDog/documentation/pull/29347), [CCM](https://docs.datadoghq.com/cloud_cost_management/recommendations/), [DBM](https://docs.datadoghq.com/database_monitoring/recommendations/)?

Author: I've pushed for that in the past but many of these insights are missing a vital piece to make then a Recommendation which is: always having a clear impact statement. 

We can say that the findings here can help with a given issue, but it's hard to definitely tie that to an impact (example: fixing GC issues will reduce CPU load but we can't concretely say by how much due to many other factors). However, when we do have that info, we promote that insight and it will show up in APM recommendations

    - name: GC Pauses
      severity: Warn
      description: Triggers if more than 10% of time was spent in GC pauses.
    - name: GC Pause Peak Duration
      severity: Info
      description: Triggers if at least one GC pause took more than 1 second.
    - name: Deadlocked Threads
      severity: Warn
      description: Triggers if we see a custom datadog event capturing deadlocked threads during the profile.
    - name: Deadlocked Threads Detected
      severity: Warn
      description: Triggers if max number of deadlocked threads over query context is bigger than 0.
    - name: Thrown Exceptions
      severity: Warn
      description: Triggers when the rate of thrown (caught and uncaught) exceptions per minute goes above a threshold (defauls to 10K)
    - name: Primitive Value Boxing
      severity: Info
      description: Triggers if more than 5% of CPU time was spent doing primitive<>object value conversions.
    - name: Explicit GC
      severity: Info
      description: Triggers if there are System.gc() calls.
    - name: Head of line blocking
      severity: Info
      description: Triggers if a queue event gets stuck behind the given activity.
---

## Overview

> Automated Analysis helps teams detect and troubleshoot problems faster, without requiring deep expertise in code profiling. Powered by Continuous Profiler, Automated Analysis continuously monitors your applications and surfaces critical issues in real time, along with actionable insights to guide resolution. When a problem is detected, Automated Analysis provides:

Reviewer: ```suggestion
Automated Analysis automatically detects performance issues in your applications using Continuous Profiler data and provides actionable insights for resolution. When an issue is detected, Automated Analysis provides:
```

- A high-level summary explaining the issue and why it matters

> - Contextual insights from profiling data (e.g., affected methods, packages, or processes)

Reviewer: ```suggestion
- Contextual insights from profiling data (for example, affected methods, packages, or processes)
```

> - Recommended next steps to help you resolve the issue efficiently

Reviewer: ```suggestion
- Recommended next steps to help you resolve the issue
```

> By reducing the expertise required to take action, Automated Analysis enables faster resolution of performance issues, increases product adoption, and helps teams avoid costly blind spots. It also lays the foundation for powering performance and cost insights across the Datadog platform.

Reviewer: ```suggestion
This reduces the profiling expertise needed to identify and resolve performance issues in your applications that might otherwise go unnoticed.
```

Author: added this on my end!

{{< img src="profiler/profiling_automated_analysis.png" alt="The Profiler Thread Time line showing a Thrown Exception insight" style="width:100%;" >}}

## Explore insights

> Automated Analysis will be surfaced within the [Profile explorer][1]

Reviewer: ```suggestion
Access Automated Analysis from the [Profile explorer][1]. Insights are displayed:
```

- As a banner at the top of the page when you're scoped to a specific service

> {{< img src="profiler/profiling_automated_analysis_banner.png" alt="The Automated Analysis banner displaying insights detected for a given service" style="width:100%;">}}

Reviewer: Note: Let's crop out the sidebar or blur your name from the screenshots.

Reviewer: Pushed commit here! https://github.com/DataDog/documentation/pull/29406/commits/1ec19bb3929df03bc68733c50e748742ccfcdee3
